app/src/main/java/org/openimis/imispolicies/util/JsonUtils.java

```diff
@@ -49,4 +59,106 @@ public static String getStringOrDefault(@NonNull JSONObject object, @NonNull Str
             return defaultValue;
         }
     }
+
+    @Nullable
+    public static Date getDateOrDefault(@NonNull JSONObject object, @NonNull String field) {
+        return getDateOrDefault(object, field, null);
+    }
+
+    @Nullable
+    public static Date getDateOrDefault(@NonNull JSONObject object, @NonNull String field, @Nullable Date defaultValue) {
+        return getDateOrDefault(object, field, defaultValue, true);
+    }
+
+    @Nullable
+    public static Date getDateOrDefault(@NonNull JSONObject object, @NonNull String field, @Nullable Date defaultValue, boolean checkNullString) {
+        String string = getStringOrDefault(object, field, null, checkNullString);
+        if (string == null) {
+            return defaultValue;
+        }
+        try {
+            return DateUtils.dateFromString(string);
+        } catch (ParseException e) {
+            return defaultValue;
+        }
+    }
+
+    @Nullable
+    public static Integer getIntegerOrDefault(@NonNull JSONObject object, @NonNull String field) {
+        return getIntegerOrDefault(object, field, null);
+    }
+
+    @Nullable
+    public static Integer getIntegerOrDefault(@NonNull JSONObject object, @NonNull String field, Integer defaultValue) {
+        return getIntegerOrDefault(object, field, defaultValue, true);
+    }
+
+    @Nullable
+    public static Integer getIntegerOrDefault(@NonNull JSONObject object, @NonNull String field, Integer defaultValue, boolean checkNullString) {
+        try {
+            if (!isStringEmpty(object, field, checkNullString)) {
+                return Integer.parseInt(object.getString(field));
+            } else {
+                return defaultValue;
+            }
+        } catch (JSONException e) {
+            return defaultValue;
+        }
+    }
```

Comment:
If I read well, `isStringEmpty` doesn't throw any exception. So, I'd do the following:

```suggestion
        if (isStringEmpty(object, field, checkNullString)) return defaultValue;
        try {
            return Integer.parseInt(object.getString(field));
        } catch (JSONException e) {
            return defaultValue;
        }
    }
```

Reply:
it applies to the rest of functions following the same pattern.

Reply:
I **never** put a return without brackets. That's a recipe for disaster. Therefore, that would not make the code a lot more readable.

Reply:
version with brackets :stuck_out_tongue: 

```java
if (isStringEmpty(object, field, checkNullString)) {return defaultValue};
```

Reply:
:joy: 

Sorry but brackets means spaces and new lines :grin: 

(I know it's a joke)